src/entity/user/UserJobPreferences.ts

```diff
@@ -0,0 +1,43 @@
+import { Column, Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';
+import { User } from './User';
+
+export enum WorkLocationType {
+  Remote = 'remote',
+  Hybrid = 'hybrid',
+  OnSite = 'on_site',
+}
+
+export interface UserCompensation {
+  currency: string;
+  amount: number;
+}
+
+@Entity()
+export class UserJobPreferences {
+  @PrimaryColumn()
+  userId: string;
+
+  @OneToOne(() => User, (user) => user.jobPreferences)
+  @JoinColumn({ name: 'userId' })
+  user: Promise<User>;
+
+  @Column({ default: false })
+  openToOpportunities: boolean;
+
+  @Column({ type: 'jsonb', default: () => [] })
+  preferredRoles: string[]; // Array of job titles
+
+  @Column({
+    nullable: true,
+  })
+  preferredLocationType: WorkLocationType;
+
+  @Column({ default: false })
+  openToRelocation: boolean;
+
+  // !! Never send this field to FE, is only stored for better recommendation !!
```

Comment:
this is controlled on gql type, so make sure its not included there so clients can't request it